src/shared/components/copyDownloadControls/ICopyDownloadControls.ts

```diff
@@ -20,4 +20,7 @@ export interface ICopyDownloadInputsProps {
     downloadLabel?: string;
     handleDownload?: () => void;
     handleCopy?: () => void;
+    // expose downloadData() to allow button to handle the data on it's own.
+    // TECH_DOWNLOADDATA: CopyDownloadButtons.downloadData needs to be async so it can work with either async context (IAsyncCopyDownloadControlsProps) or synchronous context (SimpleCopyDownloadControls)
+    downloadData?: Promise<string>;
```

Comment:
See comment in CopyDownloadControls.tsx about why downloadData is a Promise<string>